tmt/utils.py

```diff
@@ -2641,3 +2654,94 @@ def load_schema_store() -> SchemaStore:
         raise FileError(f"Failed to discover schema files\n{exc}")
 
     return store
+
+
+# A type for callbacks given to wait()
+WaitCheckType = Callable[[], T]
+
+
+def wait(
+    parent: Common,
+    check: WaitCheckType[T],
+    timeout: datetime.timedelta,
+    tick: float = DEFAULT_WAIT_TICK,
+    tick_increase: float = DEFAULT_WAIT_TICK_INCREASE
+) -> T:
+    """
+    Wait for a condition to become true.
+
+    To test the condition state, a ``check`` callback is called every ``tick``
+    seconds until ``check`` reports a success. The callback may:
+
+    * decide the condition has been fulfilled. This is a successfull outcome,
+      ``check`` shall then simply return, and waiting ends. Or,
+    * decide more time is needed. This is not a successfull outcome, ``check``
+      shall then raise :py:clas:`WaitingIncomplete` exception, and ``wait()``
+      will try again later.
+
+    :param parent: "owner" of the wait process. Used for its logging capability.
+    :param check: a callable responsible for testing the condition. Accepts no
+        arguments. To indicate more time and attempts are needed, the callable
+        shall raise :py:class:`WaitingIncomplete`, otherwise it shall return
+        without exception. Its return value will be propagated by ``wait()`` up
+        to ``wait()``'s. All other exceptions raised by ``check`` will propagate
+        to ``wait()``'s caller as well, terminating the wait.
+    :param timeout: amount of time ``wait()`` is alowed to spend waiting for
+        successfull outcome of ``check`` call.
+    :param tick: how many seconds to wait between two consecutive calls of
+        ``check``.
+    :param tick_increase: a multiplier applied to ``tick`` after every attempt.
+    :returns: value returned by ``check`` reporting success.
+    :raises GeneralError: when ``tick`` is not a positive integer.
+    :raises WaitingTimedOutError: when time quota has been consumed.
+    """
+
+    if tick <= 0:
+        raise GeneralError('Tick must be a positive integer')
+
+    NOW = datetime.datetime.utcnow
+
+    deadline = datetime.datetime.utcnow() + timeout
```

Comment:
Really a nitpick but how much do we care about host time change? It can happen ;) How about `time.monotonic()` or `time.perf_counter()` instead of `datetime.datetime.utcnow` ?

Reply:
Hm, DST-related changes should be fine, because, you know, UTC, and the rest... It's true messing with host time settings could affect the wait, but that does not seem important enough to me. Do you think otherwise?

Reply:
Well, best practise for timeouts is to used those two methods because system time can go both directions. /me thinks we should finally start doing things properly and to mitigate known issues.

Reply:
Yeah, I was hoping to distract you, but you're right. I'd say `perf_counter()` would be an overkill, `monitonic()` it is then.

Reply:
Fixed in [53cbd7e](https://github.com/teemtee/tmt/pull/1280/commits/53cbd7e41be5ae04465fc4f55bb7f28bf8b58d78).